base/math.jl

```diff
@@ -18,14 +18,19 @@ export sin, cos, tan, sinh, cosh, tanh, asin, acos, atan,
        hankelh1, hankelh2, hankelh1x, hankelh2x,
        besseli, besselix, besselk, besselkx, besselh,
        beta, lbeta, eta, zeta, polygamma, invdigamma, digamma, trigamma,
-       erfinv, erfcinv, @evalpoly
+       erfinv, erfcinv, @evalpoly,
+       acos_fast, acosh_fast, asin_fast, asinh_fast, atan2_fast,
```

Comment:
I considered this. My argument against this is that exporting them allows others to overload these functions, e.g. if you defint `sqrt(Quaternion)`. By implementing `sqrt_fast(Quaternion)` as well, you can benefit from the fastmath mechanism.

At the point where the macro `@fastmath` runs, names have not yet been resolved, so all it sees is a generic `sqrt` function. Replacing this by `Base.sqrt_fast` seems restrictive. (I'm also not sure how to implement the current fallback mechanism in `Base.sqrt_fast` so that it finds possible external `sqrt` functions, but this should be feasible.)